### PR TITLE
isabelle: 2023 -> 2024

### PR DESCRIPTION
**Pull request by jvanbruegge**
## Description of changes

https://isabelle.in.tum.de/dist/Isabelle2024/doc/NEWS.html

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by nicoonoclaste — APPROVED** [review]

**Comment by jvanbruegge** [comment]
I also just saw that the linter was updated yesterday

**Comment by jvanbruegge** [comment]
Both linux architectures work now, but darwin just says `I/O error`. I would go ahead and just mark darwin as broken. I don't have a mac and cannot really debug this

**Comment by jvanbruegge** [comment]
@wegank the merge conflict is fixed and this is ready to be merged

File: pkgs/by-name/is/isabelle/package.nix [hunk 1]
```diff
@@ -220,7 +219,7 @@ in stdenv.mkDerivation (finalAttrs: rec {
     ];
     license = licenses.bsd3;
     maintainers = [ maintainers.jwiegley maintainers.jvanbruegge ];
-    platforms = platforms.unix;
+    platforms = platforms.linux;
```
Comment (wegank) [inline pkgs/by-name/is/isabelle/package.nix:222]:
```suggestion
    platforms = platforms.unix;
    broken = stdenv.isDarwin;
```

File: pkgs/by-name/is/isabelle/package.nix [hunk 2]
```diff
@@ -130,31 +131,29 @@ in stdenv.mkDerivation (finalAttrs: rec {
...
+    for f in contrib/*/$arch/{z3,nunchaku,spass,zipperposition}; do
       patchelf --set-interpreter $(cat ${stdenv.cc}/nix-support/dynamic-linker) "$f"${lib.optionalString stdenv.isAarch64 " || true"}
     done
-    patchelf --set-interpreter $(cat ${stdenv.cc}/nix-support/dynamic-linker) contrib/bash_process-*/platform_$arch/bash_process
+    patchelf --set-interpreter $(cat ${stdenv.cc}/nix-support/dynamic-linker) contrib/bash_process-*/$arch/bash_process
     for d in contrib/kodkodi-*/jni/$arch; do
       patchelf --set-rpath "${lib.concatStringsSep ":" [ "${java}/lib/openjdk/lib/server" "${stdenv.cc.cc.lib}/lib" ]}" $d/*.so
     done
-    patchelf --set-rpath "${stdenv.cc.cc.lib}/lib" contrib/z3-*/$arch/z3
+    patchelf --set-rpath "${stdenv.cc.cc.lib}/lib" contrib/z3-*/$arch/z3${lib.optionalString stdenv.isAarch64 " || true"}
```
Comment (wegank) [inline pkgs/by-name/is/isabelle/package.nix:156]: Why `|| true` on aarch64-linux?

File: pkgs/by-name/is/isabelle/package.nix [hunk 3]
```diff
@@ -130,31 +131,29 @@ in stdenv.mkDerivation (finalAttrs: rec {
...
   '' + lib.optionalString (stdenv.hostPlatform.system == "x86_64-darwin") ''
     substituteInPlace lib/scripts/isabelle-platform \
-      --replace 'ISABELLE_APPLE_PLATFORM64=arm64-darwin' ""
+      --replace-fail 'ISABELLE_APPLE_PLATFORM64=arm64-darwin' ""
   '' + lib.optionalString stdenv.isLinux ''
-    arch=${if stdenv.hostPlatform.system == "x86_64-linux" then "x86_64-linux"
-           else if stdenv.hostPlatform.isx86 then "x86-linux"
-           else "arm64-linux"}
-    for f in contrib/*/$arch/{z3,epclextract,nunchaku,SPASS,zipperposition}; do
+    arch=${if stdenv.hostPlatform.system == "x86_64-linux" then "x86_64-linux" else "arm64-linux"}
```
Comment (wegank) [inline pkgs/by-name/is/isabelle/package.nix:148]:
```suggestion
    arch=${if stdenv.hostPlatform.system == "aarch64-linux" then "arm64-linux" else stdenv.hostPlatform.system}
```

**Comment by jvanbruegge** [comment]
@wegank all done

**Review by wegank — APPROVED** [review]

**Comment by jvanbruegge** [comment]
Ofborg is also done now

**Comment by jvanbruegge** [comment]
Thanks!

**Pull request merged**